generate_workspace/src/main/java/com/google/devtools/build/workspace/maven/VersionResolver.java

```diff
@@ -56,9 +56,14 @@ String resolveVersion(String groupId, String artifactId, String versionSpec)
     }
 
     if (isInvalidRangeResult(versions)) {
-      String errorMessage =
-          messageForInvalidArtifact(groupId, artifactId, versionSpec, "Invalid Range Result");
-      throw new InvalidArtifactCoordinateException(errorMessage);
+      if (isVersionRange(versionSpec)) {
+        // Version range resolved to no artifacts.
```

Comment:
Taking a step back, let me expand on why this function was designed as it is. The basic structure is to

* take a version specification
* request all valid versions
* use some heuristic to select the version within `selectVersion`

**Example** Say we passed "4.0", the valid versions are those >= 4.0, and `selectVersion `would select 4.0, if it exists, and the oldest version if it doesn't. With a version range like [4.0, 5.2), it would obtain all versions valid within that range. Then selectVersion would select the newest version.

**Justification** Without doing this, it would be infeasible to handle version specs like `(3.0, 5.2)`, or even `[4.0, 5.2)` for that matter. Maven's heuristic is to select the highest version. Unless the upper bound is inclusive, you will need to obtain all valid versions.

Reply:
The problem is that an empty result means that you couldn't find any valid *artifacts* - and since this method is used to validate POM files, it ends up throwing an error when your parent POM doesn't have any associated artifacts (bug #61).

The correct fix for #61 is to validate POM files without using the `resolveVersion` function, but I wasn't quite ready to delve that deeply.

Reply:
Thanks for the clarification. Looking at the pom file from #61, it does not have any dependencies.  
```
<project xmlns="http://maven.apache.org/POM/4.0.0"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0
                      https://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <parent>
    <groupId>org.sonatype.oss</groupId>
    <artifactId>oss-parent</artifactId>
    <version>9</version>
  </parent>

  <groupId>com.example</groupId>
  <artifactId>example</artifactId>
  <version>0.1.0-SNAPSHOT</version>
  <packaging>pom</packaging>
</project>
```
What do you mean by parent POM not have any associated artifacts? As in, it does not have any dependencies. From what I can see, the [parent POM](https://search.maven.org/remotecontent?filepath=org/sonatype/oss/oss-parent/9/oss-parent-9.pom) exists. Although, it does not have any dependencies.

> The correct fix for #61 is to validate POM files without using the resolveVersion function, but I wasn't quite ready to delve that deeply.

Assuming reproducing your issue simply involves copying the POM file, and calling generate_workspace on the POM file, let me play around with this. I think it would be more appropriate to make the long term fix, then to introduce this hack.

Reply:
By "parent POM" I mean the POM file for the `<parent>` project. You'll notice that [it doesn't have any artifacts](http://repo1.maven.org/maven2/org/sonatype/oss/oss-parent/9/).

The problem with using validating POM files in this way is that you're [requesting the `jar` artifact from Aether](https://github.com/bazelbuild/migration-tooling/blob/d3057b11b201935b3790f20ba47f0035ebc5e0c1/generate_workspace/src/main/java/com/google/devtools/build/workspace/maven/VersionResolver.java#L73), which isn't required to exist for a valid POM file.

Reply:
Also: I'm 100% fine with this hack not merging. I implemented it solely to get unblocked. :)

Reply:
> The problem with using validating POM files in this way is that you're [requesting the jar artifact from Aether](https://github.com/bazelbuild/migration-tooling/blob/d3057b11b201935b3790f20ba47f0035ebc5e0c1/generate_workspace/src/main/java/com/google/devtools/build/workspace/maven/VersionResolver.java#L73), which isn't required to exist for a valid POM file.

I see. By using ArtifactBuilder, any POM files without artifacts will be unduly marked as invalid. 

> Also: I'm 100% fine with this hack not merging. I implemented it solely to get unblocked. :)

As I don't have an immediate solution to the fundamental problem, let's leave the PR open as a reminder of the issue.